p2p/src/protocol/cbfmgr.rs

```diff
@@ -414,15 +414,21 @@ impl<F: Filters, U: SyncFilters + Events + Wakeup + Disconnect> FilterManager<F,
         }
     }
 
-    /// Rollback filter header chain by a given number of headers.
-    pub fn rollback(&mut self, n: usize) -> Result<(), filter::Error> {
-        self.filters.rollback(n)?;
-        self.schedule_tick();
-
-        let height = self.filters.height();
+    /// Rollback filters to the given height.
+    pub fn rollback(&mut self, height: Height) -> Result<(), filter::Error> {
+        // It's possible that a rollback doesn't affect the filter chain, if the filter headers
+        // haven't caught up to the block headers when the re-org happens.
+        if height >= self.filters.height() {
+            return Ok(());
+        }
```

Comment:
In this case, we are going back to a number of blocks, but the filter has a height lower than what we are required.

What is the case? I'm missing somethings here and maybe I need some help to understand :)

Reply:
The comment above explains it pretty clearly: This can happen if for eg. a re-org happens 4 blocks from the tip, but the filters are only synced up to 7 blocks from the tip. Any re-org of depth <= 7 will not affect the filter header chain, but will affect the block header chain.